api/v1alpha1/meta_types.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+// TypedLocalObjectReference contains enough information to let you locate the
+// typed referenced object inside the same namespace.
+// This is redacted from the upstream https://pkg.go.dev/k8s.io/api/core/v1#TypedLocalObjectReference
+type TypedLocalObjectReference struct {
```

Comment:
I think separate struct is fine. its gives us more flexibility w.r.t validation.
however this validation is specific for AddonValues where it checks for secrets/configmaps and cant be used by other fields.
Can we rename it to be specific to `ValuesReference` and move it to addon_types.go to ensure it is specific for addons?

Reply:
> however this validation is specific for AddonValues where it checks for secrets/configmaps and cant be used by other fields.

Good point!